cdap-cli/src/main/java/co/cask/cdap/cli/command/schedule/UpdateTimeScheduleCommand.java

```diff
@@ -0,0 +1,109 @@
+ * Updates a schedule.
+ */
+public final class UpdateTimeScheduleCommand extends AbstractCommand {
+
+  private final ScheduleClient scheduleClient;
+
+  @Inject
+  public UpdateTimeScheduleCommand(CLIConfig cliConfig, ScheduleClient scheduleClient) {
+    super(cliConfig);
+    this.scheduleClient = scheduleClient;
+  }
+
+  @Override
+  public void perform(Arguments arguments, PrintStream printStream) throws Exception {
+    String[] scheduleIdParts = arguments.get(ArgumentName.SCHEDULE.toString()).split("\\.");
+    String programType = arguments.get(ArgumentName.PROGRAM_TYPE.toString()).toUpperCase();
+    String[] programIdParts = arguments.get(ArgumentName.PROGRAM.toString()).split("\\.");
+    String scheduleDescription = arguments.getOptional(ArgumentName.DESCRIPTION.toString(), "");
+    String cronExpression = arguments.get(ArgumentName.CRON_EXPRESSION.toString());
+    String schedulePropertiesString = arguments.getOptional(ArgumentName.SCHEDULE_PROPERTIES.toString(), "");
+    String scheduleRunConstraintsString = arguments.getOptional(ArgumentName.SCHEDULE_RUN_CONSTRAINTS.toString(), "");
+
+    if (scheduleIdParts.length < 2 || programIdParts.length < 2 || !scheduleIdParts[0].equals(programIdParts[0])) {
+      throw new CommandInputError(this);
+    }
+
+    String appId = scheduleIdParts[0];
+    String scheduleName = scheduleIdParts[1];
+    ScheduleId scheduleId = cliConfig.getCurrentNamespace().app(appId).schedule(scheduleName);
+
+    Map<String, String> constraintsMap =
+      ArgumentParser.parseMap(scheduleRunConstraintsString, ArgumentName.SCHEDULE_RUN_CONSTRAINTS.toString());
+
+    Schedules.Builder builder = Schedules.builder(scheduleName);
+    if (constraintsMap.containsKey("maxConcurrentRuns")) {
+      builder.setMaxConcurrentRuns(Integer.valueOf("maxConcurrentRuns"));
+    }
+    if (scheduleDescription != null) {
+      builder.setDescription(scheduleDescription);
+    }
+    Schedule schedule = builder.createTimeSchedule(cronExpression);
+
+    Map<String, String> programMap = ImmutableMap.of("programName", programIdParts[1],
+                                                     "programType", programType);
+    Map<String, String> propertiesMap = ArgumentParser.parseMap(schedulePropertiesString,
+                                                                ArgumentName.SCHEDULE_PROPERTIES.toString());
+
+    ScheduleInstanceConfiguration configuration =
+      new ScheduleInstanceConfiguration("TIME", schedule, programMap, propertiesMap);
+
+    scheduleClient.update(scheduleId, configuration);
+    printStream.printf("Successfully updated schedule '%s' in program '%s'\n", scheduleName, appId);
+  }
+
+  @Override
+  public String getPattern() {
+    return String.format("update time schedule <%s> for <%s> <%s> " +
+                           "[description <%s>] at <%s> [properties <%s>] [constraints <%s>]",
+                         ArgumentName.SCHEDULE, ArgumentName.PROGRAM_TYPE, ArgumentName.PROGRAM,
```

Comment:
similar to add schedule, we should only allow program type workflow, and only require the schedule name instead of the full schedule id.